github actions: download go dependency before running test

when looking at logs for a unit test I have to scroll down pages of downloading dependecy
to see the actual uni or integration tests
we shuld add a step to github actions to download the go depdendecy first
so the output be more clean
https://github.com/kubernetes/minikube/pull/6889/checks?check_run_id=486131563

```
unit test5m 24s
##[error]Process completed with exit code 2.
Run make test
which go-bindata || GO111MODULE=off GOBIN="/home/runner/go/bin" go get github.com/jteeuwen/go-bindata/...
PATH="/usr/share/rust/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin:/home/runner/go/bin" go-bindata -nomemcopy -o pkg/minikube/assets/assets.go -pkg assets deploy/addons/...
gofmt -s -w pkg/minikube/assets/assets.go
which go-bindata || GO111MODULE=off GOBIN="/home/runner/go/bin" go get github.com/jteeuwen/go-bindata/...
PATH="/usr/share/rust/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin:/home/runner/go/bin" go-bindata -nomemcopy -o pkg/minikube/translate/translations.go -pkg translate translations/...
gofmt -s -w pkg/minikube/translate/translations.go
./test.sh
= schema_check ==========================================================
go: finding github.com/mitchellh/go-ps v0.0.0-20170309133038-4fdf99ab2936
go: finding github.com/phayes/freeport v0.0.0-20180830031419-95f893ade6f2
go: finding github.com/juju/clock v0.0.0-20190205081909-9c5c9712527c
go: finding github.com/hooklift/assert v0.0.0-20170704181755-9d1defd6d214
go: finding github.com/cloudfoundry-attic/jibber_jabber v0.0.0-20151120183258-bcc4c8345a21
go: finding github.com/machine-drivers/docker-machine-driver-vmware v0.1.1
go: finding github.com/moby/hyperkit v0.0.0-20171020124204-a12cd7250bcd
go: finding github.com/Parallels/docker-machine-parallels v1.3.0
go: finding github.com/afbjorklund/go-getter v1.4.1-0.20190910175809-eb9f6c26742c
go: finding github.com/pkg/browser v0.0.0-20160118053552-9302be274faa

```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-18.04
     steps:
     - uses: actions/checkout@v2
-    - name: build binaries
+    - name: Build Binaries
       run : |
         make minikube-linux-amd64
         make e2e-linux-amd64
@@ -36,11 +36,11 @@ jobs:
     runs-on: ubuntu-18.04
     steps:
     - uses: actions/checkout@v2
-    - name: install libvirt
+    - name: Install libvirt
       run : |
         sudo apt-get update
         sudo apt-get install -y libvirt-dev
-    - name: lint
+    - name: Lint
       env:
         TESTSUITE: lintall
       run : make test
@@ -53,11 +53,11 @@ jobs:
     runs-on: ubuntu-18.04
     steps:
     - uses: actions/checkout@v2
-    - name: install libvirt
+    - name: Install libvirt
       run : |
         sudo apt-get update
         sudo apt-get install -y libvirt-dev
-    - name: unit test
+    - name: Unit Test
       env:
         TESTSUITE: unittest
       run :
@@ -85,11 +85,11 @@ jobs:
       run: |
         curl -LO https://github.com/medyagh/gopogh/releases/download/v0.1.16/gopogh-linux-amd64
         sudo install gopogh-linux-amd64 /usr/local/bin/gopogh
-    - name: Download binaries
+    - name: Download Binaries
       uses: actions/download-artifact@v1
       with:
         name: minikube_binaries
-    - name: Run integration test
+    - name: Run Integration Test
       continue-on-error: true
       # bash {0} to allow test to continue to next step. in case of
       shell: bash {0}
@@ -107,7 +107,7 @@ jobs:
         sec=$((${TIME_ELAPSED}%60))
         TIME_ELAPSED="${min} min $sec seconds "
         echo ::set-env name=TIME_ELAPSED::${TIME_ELAPSED}
-    - name: Generate html report
+    - name: Generate HTML Report
       shell: bash
       run: |
         cd minikube_binaries
@@ -153,11 +153,11 @@ jobs:
       run: |
         curl -LO https://github.com/medyagh/gopogh/releases/download/v0.1.16/gopogh-linux-amd64
         sudo install gopogh-linux-amd64 /usr/local/bin/gopogh
-    - name: Download binaries
+    - name: Download Binaries
       uses: actions/download-artifact@v1
       with:
         name: minikube_binaries
-    - name: Run integration test
+    - name: Run Integration Test
       continue-on-error: true
       # bash {0} to allow test to continue to next step. in case of
       shell: bash {0}
@@ -175,7 +175,7 @@ jobs:
         sec=$((${TIME_ELAPSED}%60))
         TIME_ELAPSED="${min} min $sec seconds "
         echo ::set-env name=TIME_ELAPSED::${TIME_ELAPSED}
-    - name: Generate html report
+    - name: Generate HTML Report
       shell: bash
       run: |
         cd minikube_binaries
@@ -215,11 +215,11 @@ jobs:
       run: |
         curl -LO https://github.com/medyagh/gopogh/releases/download/v0.1.16/gopogh-linux-amd64
         sudo install gopogh-linux-amd64 /usr/local/bin/gopogh
-    - name: Download binaries
+    - name: Download Binaries
       uses: actions/download-artifact@v1
       with:
         name: minikube_binaries
-    - name: Run integration test
+    - name: Run Integration Test
       continue-on-error: true
       # bash {0} to allow test to continue to next step. in case of
       shell: bash {0}
@@ -237,7 +237,7 @@ jobs:
         sec=$((${TIME_ELAPSED}%60))
         TIME_ELAPSED="${min} min $sec seconds "
         echo ::set-env name=TIME_ELAPSED::${TIME_ELAPSED}
-    - name: Generate html report
+    - name: Generate HTML Report
       shell: bash
       run: |
         cd minikube_binaries
@@ -277,11 +277,11 @@ jobs:
       run: |
         curl -LO https://github.com/medyagh/gopogh/releases/download/v0.1.16/gopogh-linux-amd64
         sudo install gopogh-linux-amd64 /usr/local/bin/gopogh
-    - name: Download binaries
+    - name: Download Binaries
       uses: actions/download-artifact@v1
       with:
         name: minikube_binaries
-    - name: Run integration test
+    - name: Run Integration Test
       continue-on-error: true
       # bash {0} to allow test to continue to next step. in case of
       shell: bash {0}
@@ -299,7 +299,7 @@ jobs:
         sec=$((${TIME_ELAPSED}%60))
         TIME_ELAPSED="${min} min $sec seconds "
         echo ::set-env name=TIME_ELAPSED::${TIME_ELAPSED}
-    - name: Generate html report
+    - name: Generate HTML Report
       shell: bash
       run: |
         cd minikube_binaries
@@ -354,7 +354,7 @@ jobs:
         uses: actions/download-artifact@v1
         with:
           name: minikube_binaries
-      - name: Run integration test
+      - name: Run Integration Test
         continue-on-error: true
         # bash {0} to allow test to continue to next step. in case of
         shell: bash {0}
@@ -372,7 +372,7 @@ jobs:
           sec=$((${TIME_ELAPSED}%60))
           TIME_ELAPSED="${min} min $sec seconds "
           echo ::set-env name=TIME_ELAPSED::${TIME_ELAPSED}
-      - name: Generate html report
+      - name: Generate HTML Report
         shell: bash
         run: |
           cd minikube_binaries
@@ -405,7 +405,7 @@ jobs:
     needs: [docker_ubuntu_16_04,docker_ubuntu_18_04,none_ubuntu16_04,none_ubuntu18_04,podman_ubuntu_18_04]
     runs-on: ubuntu-18.04
     steps:
-    - name: download results docker_ubuntu_16_04
+    - name: Download Results docker_ubuntu_16_04
       uses: actions/download-artifact@v1
       with:
         name: docker_ubuntu_16_04
@@ -415,7 +415,7 @@ jobs:
       run: |
         mkdir -p all_reports
         cp -r docker_ubuntu_16_04 ./all_reports/
-    - name: download results docker_ubuntu_18_04
+    - name: Download Results docker_ubuntu_18_04
       uses: actions/download-artifact@v1
       with:
         name: docker_ubuntu_18_04
@@ -425,7 +425,7 @@ jobs:
       run: |
         mkdir -p all_reports
         cp -r docker_ubuntu_18_04 ./all_reports/
-    - name: download results none_ubuntu16_04
+    - name: Download Results none_ubuntu16_04
       uses: actions/download-artifact@v1
       with:
         name: none_ubuntu16_04
@@ -435,21 +435,21 @@ jobs:
       run: |
         mkdir -p all_reports
         cp -r none_ubuntu16_04 ./all_reports/
-    - name: download results none_ubuntu18_04
+    - name: Download Results none_ubuntu18_04
       uses: actions/download-artifact@v1
       with:
         name: none_ubuntu18_04
-    - name: cp none_ubuntu18_04 to all_report
+    - name: Copy none_ubuntu18_04 to all_report
       continue-on-error: true
       shell: bash {0}
       run: |
         mkdir -p all_reports
         cp -r none_ubuntu18_04 ./all_reports/
-    - name: download results podman_ubuntu_18_04
+    - name: Download Results podman_ubuntu_18_04
       uses: actions/download-artifact@v1
       with:
         name: podman_ubuntu_18_04
-    - name: cp podman_ubuntu_18_04 to all_report
+    - name: Copy podman_ubuntu_18_04 to all_report
       continue-on-error: true
       shell: bash {0}
       run: |

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -2,7 +2,7 @@ name: CI
 on: [pull_request]
 jobs:
   # Runs before all other jobs
-  # builds the minikube binaries 
+  # builds the minikube binaries
   build_minikube:
     env:
       TIME_ELAPSED: time
@@ -37,7 +37,7 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: install libvirt
-      run : | 
+      run : |
         sudo apt-get update
         sudo apt-get install -y libvirt-dev
     - name: lint
@@ -54,7 +54,7 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: install libvirt
-      run : | 
+      run : |
         sudo apt-get update
         sudo apt-get install -y libvirt-dev
     - name: unit test
@@ -170,7 +170,7 @@ jobs:
         START_TIME=$(date -u +%s)
         KUBECONFIG=$(pwd)/testhome/kubeconfig MINIKUBE_HOME=$(pwd)/testhome ./e2e-linux-amd64 -minikube-start-args=--driver=docker -test.timeout=70m -test.v -timeout-multiplier=3 -binary=./minikube-linux-amd64 2>&1 | tee ./report/testout.txt
         END_TIME=$(date -u +%s)
-        TIME_ELAPSED=$(($END_TIME-$START_TIME)) 
+        TIME_ELAPSED=$(($END_TIME-$START_TIME))
         min=$((${TIME_ELAPSED}/60))
         sec=$((${TIME_ELAPSED}%60))
         TIME_ELAPSED="${min} min $sec seconds "
@@ -184,7 +184,7 @@ jobs:
         STAT=$(gopogh -in ./report/testout.json -out ./report/testout.html -name "${JOB_NAME} ${GITHUB_REF}" -repo "${GITHUB_REPOSITORY}"  -details "${GITHUB_SHA}")  || true
         echo status: ${STAT}
         FailNum=$(echo $STAT | jq '.NumberOfFail')
-        TestsNum=$(echo $STAT | jq '.NumberOfTests')       
+        TestsNum=$(echo $STAT | jq '.NumberOfTests')
         GOPOGH_RESULT="${JOB_NAME} : completed with ${FailNum} / ${TestsNum} failures in ${TIME_ELAPSED}"
         echo ::set-env name=GOPOGH_RESULT::${GOPOGH_RESULT}
         echo ::set-env name=STAT::${STAT}
@@ -232,7 +232,7 @@ jobs:
         START_TIME=$(date -u +%s)
         KUBECONFIG=$(pwd)/testhome/kubeconfig MINIKUBE_HOME=$(pwd)/testhome sudo -E ./e2e-linux-amd64 -minikube-start-args=--driver=none -test.timeout=70m -test.v -timeout-multiplier=3 -binary=./minikube-linux-amd64 2>&1 | tee ./report/testout.txt
         END_TIME=$(date -u +%s)
-        TIME_ELAPSED=$(($END_TIME-$START_TIME)) 
+        TIME_ELAPSED=$(($END_TIME-$START_TIME))
         min=$((${TIME_ELAPSED}/60))
         sec=$((${TIME_ELAPSED}%60))
         TIME_ELAPSED="${min} min $sec seconds "
@@ -246,7 +246,7 @@ jobs:
         STAT=$(gopogh -in ./report/testout.json -out ./report/testout.html -name "${JOB_NAME} ${GITHUB_REF}" -repo "${GITHUB_REPOSITORY}"  -details "${GITHUB_SHA}")  || true
         echo status: ${STAT}
         FailNum=$(echo $STAT | jq '.NumberOfFail')
-        TestsNum=$(echo $STAT | jq '.NumberOfTests')       
+        TestsNum=$(echo $STAT | jq '.NumberOfTests')
         GOPOGH_RESULT="${JOB_NAME} : completed with ${FailNum} / ${TestsNum} failures in ${TIME_ELAPSED}"
         echo ::set-env name=GOPOGH_RESULT::${GOPOGH_RESULT}
         echo ::set-env name=STAT::${STAT}
@@ -276,7 +276,7 @@ jobs:
       shell: bash
       run: |
         curl -LO https://github.com/medyagh/gopogh/releases/download/v0.1.16/gopogh-linux-amd64
-        sudo install gopogh-linux-amd64 /usr/local/bin/gopogh    
+        sudo install gopogh-linux-amd64 /usr/local/bin/gopogh
     - name: Download binaries
       uses: actions/download-artifact@v1
       with:
@@ -294,7 +294,7 @@ jobs:
         START_TIME=$(date -u +%s)
         KUBECONFIG=$(pwd)/testhome/kubeconfig MINIKUBE_HOME=$(pwd)/testhome sudo -E ./e2e-linux-amd64 -minikube-start-args=--driver=none -test.timeout=70m -test.v -timeout-multiplier=3 -binary=./minikube-linux-amd64 2>&1 | tee ./report/testout.txt
         END_TIME=$(date -u +%s)
-        TIME_ELAPSED=$(($END_TIME-$START_TIME)) 
+        TIME_ELAPSED=$(($END_TIME-$START_TIME))
         min=$((${TIME_ELAPSED}/60))
         sec=$((${TIME_ELAPSED}%60))
         TIME_ELAPSED="${min} min $sec seconds "
@@ -308,7 +308,7 @@ jobs:
         STAT=$(gopogh -in ./report/testout.json -out ./report/testout.html -name "${JOB_NAME} ${GITHUB_REF}" -repo "${GITHUB_REPOSITORY}"  -details "${GITHUB_SHA}")  || true
         echo status: ${STAT}
         FailNum=$(echo $STAT | jq '.NumberOfFail')
-        TestsNum=$(echo $STAT | jq '.NumberOfTests')       
+        TestsNum=$(echo $STAT | jq '.NumberOfTests')
         GOPOGH_RESULT="${JOB_NAME} : completed with ${FailNum} / ${TestsNum} failures in ${TIME_ELAPSED}"
         echo ::set-env name=GOPOGH_RESULT::${GOPOGH_RESULT}
         echo ::set-env name=STAT::${STAT}
@@ -343,8 +343,8 @@ jobs:
           sudo apt-key add - < Release.key || true
           sudo apt-get update -qq
           sudo apt-get -qq -y install podman
-          sudo podman version || true 
-          sudo podman info || true 
+          sudo podman version || true
+          sudo podman info || true
       - name: Install gopogh
         shell: bash
         run: |
@@ -367,7 +367,7 @@ jobs:
           START_TIME=$(date -u +%s)
           KUBECONFIG=$(pwd)/testhome/kubeconfig MINIKUBE_HOME=$(pwd)/testhome sudo -E ./e2e-linux-amd64 -minikube-start-args=--driver=podman -test.timeout=70m -test.v -timeout-multiplier=3 -binary=./minikube-linux-amd64 2>&1 | tee ./report/testout.txt
           END_TIME=$(date -u +%s)
-          TIME_ELAPSED=$(($END_TIME-$START_TIME)) 
+          TIME_ELAPSED=$(($END_TIME-$START_TIME))
           min=$((${TIME_ELAPSED}/60))
           sec=$((${TIME_ELAPSED}%60))
           TIME_ELAPSED="${min} min $sec seconds "
@@ -381,7 +381,7 @@ jobs:
           STAT=$(gopogh -in ./report/testout.json -out ./report/testout.html -name "${JOB_NAME} ${GITHUB_REF}" -repo "${GITHUB_REPOSITORY}"  -details "${GITHUB_SHA}")  || true
           echo status: ${STAT}
           FailNum=$(echo $STAT | jq '.NumberOfFail')
-          TestsNum=$(echo $STAT | jq '.NumberOfTests')       
+          TestsNum=$(echo $STAT | jq '.NumberOfTests')
           GOPOGH_RESULT="${JOB_NAME} : completed with ${FailNum} / ${TestsNum} failures in ${TIME_ELAPSED}"
           echo ::set-env name=GOPOGH_RESULT::${GOPOGH_RESULT}
           echo ::set-env name=STAT::${STAT}
@@ -398,7 +398,7 @@ jobs:
           echo $STAT | jq '.FailedTests' || true
           echo "-------------------------------------------------------"
           if [ "$numFail" -gt 0 ];then echo "*** $numFail Failed ***";exit 2;fi
-  # After all 4 integration tests finished 
+  # After all 4 integration tests finished
   # collect all the reports and upload
   upload_all_reports:
     if: always()

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -1,5 +1,7 @@
 name: CI
 on: [pull_request]
+env:
+  GOPROXY: https://proxy.golang.org
 jobs:
   # Runs before all other jobs
   # builds the minikube binaries
@@ -11,6 +13,8 @@ jobs:
     runs-on: ubuntu-18.04
     steps:
     - uses: actions/checkout@v2
+    - name: Download Dependencies
+      run : go mod download
     - name: Build Binaries
       run : |
         make minikube-linux-amd64
@@ -40,6 +44,8 @@ jobs:
       run : |
         sudo apt-get update
         sudo apt-get install -y libvirt-dev
+    - name: Download Dependencies
+      run : go mod download
     - name: Lint
       env:
         TESTSUITE: lintall
@@ -57,6 +63,8 @@ jobs:
       run : |
         sudo apt-get update
         sudo apt-get install -y libvirt-dev
+    - name: Download Dependencies
+      run : go mod download
     - name: Unit Test
       env:
         TESTSUITE: unittest

